sound/soc/sof/intel/hda-loader.c

```diff
@@ -34,6 +34,8 @@
 #include "../ops.h"
 #include "hda.h"
 
+#define HDA_FW_BOOT_ATTEMPTS	3
```

Comment:
any idea why 3 is good? 
Also how was this tested?

Reply:
3 comes from legacy but not sure if that is the official recommendation. Do you have any pointers?

Reply:
> any idea why 3 is good?
> Also how was this tested?

I'm not sure how to test it because so far fw seems to boot at the first try. Maybe I can artificially add a failure after the first iteration?

Reply:
I don't know why legacy chose 3.
I think the main point is that we need to have a test harness where we boot hundreds/thousands of times, check if we have any issues and measure the benefits of this retry loop. I vaguely recall seeing this for ApolloLake, but not sure how it was tested.

Reply:
@plbossart i think that makes sense. The suspend_stress_test that we perform today for about 2500 cycles essentially does this by powering off the DSP and booting the FW again (and going beyond that). We have not come across issues with FW boot with this test so far even without the logic to retry FW boot.